api/preprints/permissions.py

```diff
@@ -17,6 +17,8 @@ def has_object_permission(self, request, view, obj):
             if auth.user is None:
                 return obj.verified_publishable
             else:
+                if obj.provider.reviews_workflow:
+                    return obj.verified_publishable or (node.is_public and auth.user.has_perm('view_submissions', obj.provider)) or node.has_permission(auth.user, osf_permissions.ADMIN) or (node.is_contributor(auth.user) and obj.reviews_state != 'initial')
```

Comment:
Minor: I don't like how much is duplicated between this and the line below. Could you do something like assign the part in common to a variable and `return foo or (node.is_contributor(auth.user) and obj.reviews_state != 'initial')` here?